File: 1436. Destination City - Leetcode/1436. Destination City.java
class Solution {
    public String destCity(List<List<String>> paths) {
        HashSet<String> dest = new HashSet();
        for (int i = 0; i < paths.size(); i++) {
            dest.add(paths.get(i).get(0));
        }
        
        for (int i = 0; i < paths.size(); i++) {
            String finaldest = paths.get(i).get(1);
            if (!dest.contains(finaldest)) {
                return finaldest;
            }
        } 
        return "";
    }
}

File: 1436. Destination City - Leetcode/1436. Destination City.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_set<std::string> dest;
        
        for (int i = 0; i < paths.size(); i++) {
            dest.insert(paths[i][0]);
        }

        for (int i = 0; i < paths.size(); i++) {
           string finalDest = paths[i][1];
            if (dest.find(finalDest) == dest.end()) {
                return finalDest;
            }
        }

        return "";
    }
};
